pkg/controller/baremetalhost/baremetalhost_controller_test.go

```diff
@@ -184,6 +184,24 @@ func waitForProvisioningState(t *testing.T, r *ReconcileBareMetalHost, host *met
 	)
 }
 
+// TestPause ensures that the requeue happens when the pause annotation is there.
+func TestPause(t *testing.T) {
+	host := newDefaultHost(t)
+	host.Annotations = map[string]string{
+		metal3v1alpha1.PausedAnnotation: "true",
+	}
+	r := newTestReconciler(host)
+
+	tryReconcile(t, r, host,
+		func(host *metal3v1alpha1.BareMetalHost, result reconcile.Result) bool {
+			if result.Requeue && result.RequeueAfter == pauseRetryDelay {
```

Comment:
It would be good to test that some action taken early by reconcile has not been taken for this host, to prove not just that Reconcile() returned the right RequeueAfter value but that it is in fact paused.

Reply:
I have added a check on the finalizers that are the next step. There should not be any finalizer.